daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -991,12 +999,14 @@ object SBuiltin {
                   id.packageId,
                   pkg => {
                     machine.compiledPackages = pkg
-                    machine.ctrl = translateValue(machine, value)
+                    importValue(machine, value)
                   }
                 ))
           }
       }
-    CtrlValue(go(value))
+
+    machine.addGlobalContractIds(cids.result())
```

Comment:
This looks fishy. You're calling `go`, which modifies `cids`, after you're getting the `result` out.

Reply:
indeed. :(